samza-core/src/main/java/org/apache/samza/storage/blobstore/util/BlobStoreUtil.java

```diff
@@ -629,4 +637,15 @@ private static Predicate<Throwable> isCauseNonRetriable() {
       return unwrapped != null && !RetriableException.class.isAssignableFrom(unwrapped.getClass());
     };
   }
+
+  private RetryPolicy<Object> buildRetryPolicyFromConfig() {
```

Comment:
Consider moving this method to BlobStoreConfigs and returning a RetryPolicy/RetryPolicyConfig directly (i.e. instead of returning the configs individually through multiple accessors and constructing this object later).